clang-11 build is broken (from 2a7cfaa?)

We've did a mistake switching from clang-10 to clang-11 in CI and accidentally disabled clang build at all (which by fact was done thru gcc while marked as clang-11). As a result build regression squeezed in. See https://github.com/Intel-Media-SDK/product-configs/pull/364 for build result (this PR enables clang-11 for real). Presumably regression started from 2a7cfaa.

```
/docker/worker/mediasdk-clang-11_0/build_dir/repos/MediaSDK/_studio/shared/src/libmfx_core_factory.cpp:42: undefined reference to `VAAPIVideoCORE_T<CommonCORE>::VAAPIVideoCORE_T(unsigned int, unsigned int, _mfxSession*)'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
_studio/mfx_lib/plugin/CMakeFiles/mfx_h264la_hw64.dir/build.make:679: recipe for target '__bin/release/libmfx_h264la_hw64.so' failed
make[2]: *** [__bin/release/libmfx_h264la_hw64.so] Error 1
make[2]: Leaving directory '/docker/worker/mediasdk-clang-11_0/build_dir/build'
CMakeFiles/Makefile2:4511: recipe for target '_studio/mfx_lib/plugin/CMakeFiles/mfx_h264la_hw64.dir/all' failed
make[1]: *** [_studio/mfx_lib/plugin/CMakeFiles/mfx_h264la_hw64.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[100%] Linking CXX shared library ../../__bin/release/libmfxhw64.so
cd /docker/worker/mediasdk-clang-11_0/build_dir/build/_studio/mfx_lib && /usr/bin/cmake -E cmake_link_script CMakeFiles/mfxhw64.dir/link.txt --verbose=1
/usr/bin/clang++-11 -fPIC -DMFX_TRACE_ENABLE_STAT -DMFX_TRACE_ENABLE_TEXTLOG -std=c++11 -m64 -g  -pipe -fPIC -Wall -Wformat -Wformat-security -Wnon-virtual-dtor -Wno-deprecated-declarations -Wno-unk
nown-pragmas -Wno-unused  -DUNIX -D__USE_LARGEFILE64 -D_FILE_OFFSET_BITS=64 -DLINUX -DLINUX32 -DLINUX64  -DMFX_BUILD_INFO="\"Ubuntu 4.15.0-117-generic | Clang 11.0.0 | glibc 2.27\"" -DMFX_API_VERSION="\"1.34+\"" -DMFX_GIT_COMMIT="\"2a7cfaa4\"" -DMEDIA_VERSION_STR="\"20.4.0\"" -DMFX_TRACE_ENABLE_ITT -O2 -Wformat -Wformat-security -Wall -Werror -D_FORTIFY_SOURCE=2 -fstack-protector-strong -L/docker/worker/mediasdk-clang-11_0/build_dir/dependencies/libva/usr/local/lib64 -Wl,--no-undefined,-z,relro,-z,now,-z,noexecstack -Wl,--no-as-needed -ldl  -Wl,--version-script=/docker/worker/mediasdk-clang-11_0/build_dir/repos/MediaSDK/_studio/mfx_lib/libmfxhw.map -fstack-protector   -Wl,--no-undefined,-z,relro,-z,now,-z,noexecstack -fstack-protector  -shared -Wl,-soname,libmfxhw64.so.1 -o ../../__bin/release/libmfxhw64.so.1.34 CMakeFiles/mfxhw64.dir/scheduler/linux/src/mfx_scheduler_core.cpp.o CMakeFiles/mfxhw64.dir/scheduler/linux/src/mfx_scheduler_core_ischeduler.cpp.o CMakeFiles/mfxhw64.dir/scheduler/linux/src/mfx_scheduler_core_iunknown.cpp.o CMakeFiles/mfxhw64.dir/scheduler/linux/src/mfx_scheduler_core_task.cpp.o CMakeFiles/mfxhw64.dir/scheduler/linux/src/mfx_scheduler_core_task_management.cpp.o CMakeFiles/mfxhw64.dir/scheduler/linux/src/mfx_scheduler_core_thread.cpp.o CMakeFiles/mfxhw64.dir/shared/src/libmfxsw.cpp.o CMakeFiles/mfxhw64.dir/shared/src/libmfxsw_async.cpp.o CMakeFiles/mfxhw64.dir/shared/src/libmfxsw_decode.cpp.o CMakeFiles/mfxhw64.dir/shared/src/libmfxsw_enc.cpp.o CMakeFiles/mfxhw64.dir/shared/src/libmfxsw_encode.cpp.o CMakeFiles/mfxhw64.dir/shared/src/libmfxsw_pak.cpp.o CMakeFiles/mfxhw64.dir/shared/src/libmfxsw_plugin.cpp.o CMakeFiles/mfxhw64.dir/shared/src/libmfxsw_query.cpp.o CMakeFiles/mfxhw64.dir/shared/src/libmfxsw_session.cpp.o CMakeFiles/mfxhw64.dir/shared/src/libmfxsw_vpp.cpp.o CMakeFiles/mfxhw64.dir/shared/src/mfx_session.cpp.o CMakeFiles/mfxhw64.dir/shared/src/mfx_user_plugin.cpp.o CMakeFiles/mfxhw64.dir/shared/src/mfx_brc_common.cpp.o CMakeFiles/mfxhw64.dir/shared/src/mfx_vc1_dec_common.cpp.o CMakeFiles/mfxhw64.dir/shared/src/mfx_vpx_dec_common.cpp.o CMakeFiles/mfxhw64.dir/__/shared/src/cm_mem_copy.cpp.o CMakeFiles/mfxhw64.dir/__/shared/src/fast_copy_c_impl.cpp.o CMakeFiles/mfxhw64.dir/__/shared/src/fast_copy.cpp.o CMakeFiles/mfxhw64.dir/__/shared/src/mfx_vpp_vaapi.cpp.o CMakeFiles/mfxhw64.dir/__/shared/src/libmfx_allocator.cpp.o CMakeFiles/mfxhw64.dir/__/shared/src/libmfx_allocator_vaapi.cpp.o CMakeFiles/mfxhw64.dir/__/shared/src/libmfx_core.cpp.o CMakeFiles/mfxhw64.dir/__/shared/src/libmfx_core_hw.cpp.o CMakeFiles/mfxhw64.dir/__/shared/src/libmfx_core_factory.cpp.o CMakeFiles/mfxhw64.dir/__/shared/src/libmfx_core_vaapi.cpp.o CMakeFiles/mfxhw64.dir/__/shared/src/mfx_umc_alloc_wrapper.cpp.o CMakeFiles/mfxhw64.dir/__/shared/src/mfx_umc_mjpeg_vpp.cpp.o CMakeFiles/mfxhw64.dir/__/shared/src/mfx_static_assert_structs.cpp.o CMakeFiles/mfxhw64.dir/__/shared/src/mfx_mfe_adapter.cpp.o CMakeFiles/fast_copy_sse4.dir/__/shared/src/fast_copy_sse4_impl.cpp.o  -L/opt/intel/vtune_amplifier/lib64 -Xlinker --start-group ../../__lib/release/libbitrate_control.a ../../__lib/release/libencode_sw.a ../../__lib/release/libipp.a ../../__lib/release/libencode.a ../../__lib/release/libumc.a ../../__lib/release/libvm.a ../../__lib/release/libvm_plus.a ../../__lib/release/libumc_va_hw.a ../../__lib/release/libcmrt_cross_platform_hw.a ../../__lib/release/libdecode_hw.a ../../__lib/release/libencode_hw.a ../../__lib/release/libh264_la.a ../../__lib/release/libgenx.a ../../__lib/release/libmfx_common.a ../../__lib/release/libmfx_common_hw.a ../../__lib/release/libasc.a ../../__lib/release/libmctf_hw.a ../../__lib/release/libvpp_hw.a ../../__lib/release/libh264_common.a ../../__lib/release/libh264_preenc.a ../../__lib/release/libh264_enc.a ../../__lib/release/libh264_pak.a ../../__lib/release/libmfx_trace.a -littnotify -lpthread -ldl -Xlinker --end-group -lva -lgcc 
../../__lib/release/libencode_hw.a(mfx_h264_encode_vaapi.cpp.o): In function `MfxHwH264Encode::VAAPIEncoder::CreateAuxilliaryDevice(VideoCORE*, GUID, unsigned int, unsigned
 int, bool)':
/docker/worker/mediasdk-clang-11_0/build_dir/repos/MediaSDK/_studio/mfx_lib/shared/src/mfx_h264_encode_vaapi.cpp:1468: undefined reference to `VAAPIVideoCORE_T<CommonCORE>::GetVAService(void**)'
CMakeFiles/mfxhw64.dir/__/shared/src/mfx_vpp_vaapi.cpp.o: In function `MfxHwVideoProcessing::VAAPIVideoProcessing::CreateDevice(VideoCORE*, mfxVideoParam*, bool)':
/docker/worker/mediasdk-clang-11_0/build_dir/repos/MediaSDK/_studio/shared/src/mfx_vpp_vaapi.cpp:130: undefined reference to `VAAPIVideoCORE_T<CommonCORE>::GetVAService(void**)'
CMakeFiles/mfxhw64.dir/__/shared/src/libmfx_core_factory.cpp.o: In function `FactoryCORE::CreateCORE(eMFXVAType, unsigned int, unsigned int, _mfxSession*)':
/docker/worker/mediasdk-clang-11_0/build_dir/repos/MediaSDK/_studio/shared/src/libmfx_core_factory.cpp:42: undefined reference to `VAAPIVideoCORE_T<CommonCORE>::VAAPIVideoCORE_T(unsigned int, unsigned int, _mfxSession*)'
../../__lib/release/libdecode_hw.a(mfx_vp9_dec_decode_hw.cpp.o): In function `VideoDECODEVP9_HW::PrepareInternalSurface(int&, mfxFrameInfo&)':
/docker/worker/mediasdk-clang-11_0/build_dir/repos/MediaSDK/_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp:896: undefined reference to `VAAPIVideoCORE_T<CommonCORE>::ReallocFrame(mfxFrameSurface1*)'
../../__lib/release/libencode_hw.a(mfx_mpeg2_encode_vaapi.cpp.o): In function `MfxHwMpeg2Encode::VAAPIEncoder::CreateAuxilliaryDevice(unsigned short)':
/docker/worker/mediasdk-clang-11_0/build_dir/repos/MediaSDK/_studio/mfx_lib/shared/src/mfx_mpeg2_encode_vaapi.cpp:374: undefined reference to `VAAPIVideoCORE_T<CommonCORE>::GetVAService(void**)'
../../__lib/release/libencode_hw.a(mfx_mpeg2_encode_vaapi.cpp.o): In function `MfxHwMpeg2Encode::VAAPIEncoder::Init(tagENCODE_FUNC, MfxHwMpeg2Encode::ExecuteBuffers*)':
/docker/worker/mediasdk-clang-11_0/build_dir/repos/MediaSDK/_studio/mfx_lib/shared/src/mfx_mpeg2_encode_vaapi.cpp:501: undefined reference to `VAAPIVideoCORE_T<CommonCORE>::GetVAService(void**)'
../../__lib/release/libencode_hw.a(mfx_mjpeg_encode_vaapi.cpp.o): In function `MfxHwMJpegEncode::VAAPIEncoder::CreateAuxilliaryDevice(VideoCORE*, unsigned int, unsigned int, bool)':
/docker/worker/mediasdk-clang-11_0/build_dir/repos/MediaSDK/_studio/mfx_lib/encode_hw/mjpeg/src/mfx_mjpeg_encode_vaapi.cpp:73: undefined reference to `VAAPIVideoCORE_T<CommonCORE>::GetVAService(void**)'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/_studio/shared/src/libmfx_core_vaapi.cpp
+++ b/_studio/shared/src/libmfx_core_vaapi.cpp
@@ -439,8 +439,6 @@ mfx_device_item getDeviceItem(VADisplay pVaDisplay)
     return retDeviceItem;
 } // eMFXHWType getDeviceItem (VADisplay pVaDisplay)
 
-template class VAAPIVideoCORE_T<CommonCORE  >;
-
 template <class Base>
 VAAPIVideoCORE_T<Base>::VAAPIVideoCORE_T(
     const mfxU32 adapterNum,
@@ -1510,6 +1508,8 @@ void* VAAPIVideoCORE_T<Base>::QueryCoreInterface(const MFX_GUID &guid)
 
     return Base::QueryCoreInterface(guid);
 } // void* VAAPIVideoCORE_T<Base>::QueryCoreInterface(const MFX_GUID &guid)
+
+template class VAAPIVideoCORE_T<CommonCORE  >;
 
 bool IsHwMvcEncSupported()
 {

